assets/src/edit-story/components/header/buttons.js

```diff
@@ -55,19 +59,94 @@ function PreviewButton() {
       meta: { isSaving },
       story: { link },
     },
+    actions: { saveStory },
   } = useStory();
 
+  const [previewLinkToOpenViaDialog, setPreviewLinkToOpenViaDialog] = useState(
+    null
+  );
+
   /**
    * Open a preview of the story in current window.
    */
   const openPreviewLink = () => {
     const previewLink = addQueryArgs(link, { preview: 'true' });
-    window.open(previewLink, 'story-preview');
+
+    // Start a about:blank popup with waiting message until we complete
+    // the saving operation. That way we will not bust the popup timeout.
+    let popup;
+    try {
+      popup = window.open('about:blank', PREVIEW_TARGET);
+      if (popup) {
+        popup.document.write('<!DOCTYPE html><html><body>');
+        // Output "waiting" message.
+        popup.document.write(
+          escapeHTML(
+            __('Please wait. Generating the preview...', 'web-stories')
+          )
+        );
+        // Force redirect to the preview URL after 5 seconds. The saving tab
+        // might get frozen by the browser.
+        popup.document.write(
+          `<script>
+            setTimeout(function() {
+              location.replace(${JSON.stringify(previewLink)});
+            }, 5000);
+          </script>`
+        );
+      }
+    } catch (e) {
+      // Ignore errors. Anything can happen with a popup. The errors
+      // will be resolved after the story is saved.
+    }
+
+    // DO NOT SUBMIT: what if the status is no longer draft?
+    saveStory({ status: 'draft' }).then(() => {
+      let previewOpened = false;
+      if (popup) {
+        try {
+          popup.location.replace(previewLink);
+          previewOpened = true;
+        } catch (e) {
+          // Ignore the errors. They will simply trigger the "try again"
+          // dialog.
+        }
+      }
+      if (!previewOpened) {
+        setPreviewLinkToOpenViaDialog(previewLink);
```

Comment:
Under which circumstances does this happen? If popups are blocked for domain? I just want to see it in action locally 😄

Reply:
A good way to test it locally is to somehow make saving slow enough so that you have time to navigate the preview window before it completes to some other domain.  It's a little weird case, but still...

Reply:
Oh, tricky... I'll try to make it happen, as I want to see this flow.

@swissspidy, in which file should I insert `sleep(10)`?

Reply:
Just pull the latest branch for this PR and wrap `saveStory.then(() => {})` content into `setTimeout` :)